src/components/atoms/SpacesDropdown/SpacesDropdown.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useMemo, useState } from "react";
+import { Dropdown as ReactBootstrapDropdown } from "react-bootstrap";
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
+
+import { ADMIN_V1_ROOMS_URL, VENUE_SPACES_ICONS_MAPPING } from "settings";
+
+import { Room } from "types/rooms";
+import { ContainerClassName } from "types/utility";
+
+import { Dropdown } from "components/atoms/Dropdown";
+
+import "./SpacesDropdown.scss";
+
+export interface SpacesDropdownProps extends ContainerClassName {
+  defaultSpace?: string;
+  venueSpaces: Room[];
+  venueId?: string;
+  setValue: <T>(prop: string, value: T, validate: boolean) => void;
+  fieldName: string;
+}
+
+export const SpacesDropdown: React.FC<SpacesDropdownProps> = ({
+  defaultSpace,
+  venueSpaces,
+  venueId,
+  containerClassName,
```

Comment:
The only place this component is used doesn't require outside class injection. This prop and the extends can safely be removed. The only use:

```tsx
                <SpacesDropdown
                  venueSpaces={venue.rooms ?? ALWAYS_EMPTY_ARRAY}
                  venueId={venueId}
                  setValue={setValue}
                  fieldName="room"
                  defaultSpace={event?.room}
                />
```

Reply:
fair. fixed in 919b1f031c1e636c9de7dbd1aabb12ce7f337fef